### ISSUE TITLE
[ERROR] InnoDB: Tried to read xyz bytes at offset xyz. Was only able to read 0.

### ISSUE THREAD

**Issue by lippoliv** [zabbix/zabbix-community-docker#88]
Hey there,
as of I have a new strategy to Setup and Backup my Servers, I wanted to re-setup my Monitoring-Server as well. I have it since some weeks. And I use the great monitoringartist docker Containers (thanks for all your work).

Let me say it first: At this Point I did not Crash any data ;)

To test if it will Startup again, and to test Docker Compose instead of an own Shell file (I had a Shell file, now I migrate all docker-projects to docker-compose) I did the following:
1. Shutdown on LIVE
2. Run incremental backup of LIVE
3. ZIP docker volumes from LIVE
4. Download
5. Write `docker-compose.yml`
6. Run locally

I thought it will run, or Crash if I had forgotten to migrate something to the docker compose file. But something else happened:

```
db_1      | 170602 05:57:18 mysqld_safe Logging to '/var/lib/mysql/error.log'.
db_1      | 170602 05:57:18 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
db_1      | 170602 05:57:18 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
db_1      | 170602  5:57:18 [Warning] options --log-slow-admin-statements, --log-queries-not-using-indexes and --log-slow-slave-statements have no effect if --log_slow_queries is not set
db_1      | 170602  5:57:18 [Note] /usr/sbin/mysqld (mysqld 10.0.31-MariaDB) starting as process 254 ...
db_1      | 170602  5:57:18 [Warning] Could not increase number of max_open_files to more than 4096 (request: 4407)
db_1      | 170602  5:57:18 [Note] InnoDB: Using mutexes to ref count buffer pool pages
db_1      | 170602  5:57:18 [Note] InnoDB: The InnoDB memory heap is disabled
db_1      | 170602  5:57:18 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
db_1      | 170602  5:57:18 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
db_1      | 170602  5:57:18 [Note] InnoDB: Compressed tables use zlib 1.2.7
db_1      | 170602  5:57:18 [Note] InnoDB: Using Linux native AIO
db_1      | 170602  5:57:18 [Note] InnoDB: Using CPU crc32 instructions
db_1      | 170602  5:57:18 [Note] InnoDB: Initializing buffer pool, size = 768.0M
db_1      | 170602  5:57:18 [Note] InnoDB: Completed initialization of buffer pool
db_1      | 170602  5:57:18 [ERROR] InnoDB: Tried to read 16384 bytes at offset 81920. Was only able to read 0.
db_1      | 2017-06-02 05:57:18 7fc56a5b28c0  InnoDB: Operating system error number 22 in a file operation.
db_1      | InnoDB: Error number 22 means 'Invalid argument'.
db_1      | InnoDB: Some operating system error numbers are described at
db_1      | InnoDB: http://dev.mysql.com/doc/refman/5.6/en/operating-system-error-codes.html
db_1      | 170602  5:57:18 [ERROR] InnoDB: File (unknown): 'read' returned OS error 122. Cannot continue operation
db_1      | 170602 05:57:18 mysqld_safe mysqld from pid file /var/lib/mysql/mysql.pid ended
db_1      | Still waiting for DB service...
db_1      | Still waiting for DB service...
db_1      | Still waiting for DB service...
db_1      | Still waiting for DB service...
db_1      | Still waiting for DB service...
```
I read around and it sounds like this error occures if your filesystem is full, but on that Windows Systems HDD I have severel hundret GB free. My docker-volumes are just a couple GB.

So I started, maybe Windows is here the Problem, I uploaded that docker-compose Project to another Server and tryed to start it there (even there enough free disk space) with the same result.

Any Idea what I do wrong?

Here the `docker-compose.yml`

```
version: '3'

services:
  db:
    image: monitoringartist/zabbix-db-mariadb:latest
    restart: always
    environment:
      - MARIADB_USER=mariadbuser
      - MARIADB_PASS=mariadbpass
    volumes:
      - ./mysql:/var/lib/mysql
      - ./zabbixdb:/backups
      - /etc/localtime:/etc/localtime:ro
    
  zabbix:
    image: monitoringartist/zabbix-xxl:3.2.4
    restart: always
    environment:
      - ZS_DBUser=mariadbuser
      - ZS_DBPassword=mariadbpass
      - ZS_DBHost=db
      - XXL_analytics=false
      - XXL_apiuser=apiuser
      - XXL_apipass=apipass
      - PHP_date_timezone=Europe/Berlin
    ports:
      - "8080:80"
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./zabbix/alertscripts:/usr/local/share/zabbix/alertscripts
```

**Comment by jangaraj** [comment]
You are trying to replicate binary files (mariadb data files in the volume) between Linux and Windows. It's possible in the theory, but as you see it's not easy. There is a lot of thinks, which can be a problem: Docker storage driver, file permissions, files owner, docker sec. configs, docker configs. It's not a issue of this project, so I'm closing issue. 

My recommendation: use classic mysql dump to sql and then import that sql.

**Comment by lippoliv** [comment]
@jangaraj Did you read that I tryed it also on a Linux Server with the same result?

> So I started, maybe Windows is here the Problem, I uploaded that docker-compose Project to another Server and tryed to start it there (even there enough free disk space) with the same result.

**Comment by jangaraj** [comment]
Linux - Linux can have a same problems: Docker storage driver, file permissions, files owner, docker sec. configs, docker configs. 

Actually I rsynced my Zabbix DB data files 2 days ago (Ubuntu 14 - Ubuntu 14). DB was not able to start. Problem: DB files owner (I've used my user for ssh rsync). Owner fixed - problem solved.

There can be a million things which can be a wrong. Sorry, but don't expect to solve me your problem for free. My time has a value for me.
